src/engraving/rendering/dev/beamtremololayout.cpp

```diff
@@ -96,40 +96,68 @@ void BeamTremoloLayout::offsetBeamToRemoveCollisions(const BeamBase* item, const
 
     // tolerance eliminates all possibilities of floating point rounding errors
     const double tolerance = ldata->beamWidth * 0.25 * (ldata->up ? -1 : 1);
-    bool isSmall = ldata->isGrace || ldata->mag() < 1.;
+    const bool isSmall = ldata->isGrace || ldata->mag() < 1.;
 
     double startY = (isStartDictator ? dictator : pointer) * ldata->spatium / 4 + tolerance;
     double endY = (isStartDictator ? pointer : dictator) * ldata->spatium / 4 + tolerance;
+    // min stem lengths according to how many beams there are (starting with 1)
+    static const int minStemLengths[] = { 11, 13, 15, 18, 21, 24, 27, 30 };
```

Comment:
Couple of things about this. 
a) It is unclear at first sight in which units these values are represented. I'm guessing, from the way that you are using it later, that these are quarter-spaces? I think it would be clearer if you expressed these numbers in units of `spatium` (also good to write a comment that says so). This will also save you from doing that strange `* 4` multiplication in `desiredLen`.
b) Please make this into a function, so its meaning and use are more transparent. Something like 
double BeamTremoloLayout::minStemLength(int beamsCount) { ... }. Then define this array inside the funtion itself and that way you can also safe guard against out-of-range access by checking that `beamsCount` is within range.
c) You can make it not just `const` but `constexpr`, that way it will be evaluated at compile time.

Reply:
By the way, I know that this thing of using quarter-spaces is done a lot in the beam layout code (and it's totally fine given that beams adjust in quarter-space increments) so I wasn't critizing you for that. I just want to avoid perpetuating the bad practice of doing `* 4` and `/ 4`  all the time, which is obscure and prone to mistakes :)

Reply:
Stem lengths are calculated in quarter spaces everywhere else - ostensibly to avoid floating point errors.  I'd be inclined to leave it as is and leave a clarifying comment!